### PR TITLE
Limit Subscribe txn string length

### PR DESCRIPTION
**Pull request by gdmmx**
Signed-off-by: gdmmx <gdmmx@163.com>

### Proposed changes in this pull request
Urgent bugfix temporary: limit Subscribe txn string length

### Type (put an `x` where ever applicable)
- [x] Bug fix: Link to the issue
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Please put an `x` against the checkboxes. Write a small comment explaining if its `N/A` (not applicable)

- [x] Read the [CONTRIBUTION guidelines](https://github.com/nknorg/nkn#contributing).
- [ ] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [ ] Added proper documentation where ever applicable (in code and README.md).
- [x] Code has been written according to [NKN-Golang-Style-Guide](https://github.com/nknorg/nkn/wiki/NKN-Golang-Style-Guide)

### Extra information
Any extra information related to this pull request.

### CODE REVIEW

File: pb/transaction.go [hunk 1]
```diff
@@ -135,3 +136,24 @@ func (m *SigChainTxn) ToMap() map[string]interface{} {
 		"Submitter": common.BytesToUint160(m.Submitter),
 	}
 }
+
+type SubTxnDBChangePoint uint32
+
+const (
+	LimitMetaCP SubTxnDBChangePoint = 124000
```
Comment (NoelBright) [inline pb/transaction.go]: How about move this const SubTxnDBChangePoint to config.go?

**Comment by yilunzhang** [comment]
What about this:
in config.go add 2 constants `maxAddrIdentifierSize` and `maxSubscribeMetaSize`, and 2 methods `GetMaxAddrIdentifierSize(height)` and `GetMaxSubscribeMetaSize(height)`, and return corresponding constant depending on height. This way we are separating verification logic (chain/txValidator.go) and constant definition (config.go).

**Comment by gdmmx** [comment]
> What about this:
> in config.go we add 2 constants (for latest value) `maxAddrIdentifierSize = 64` and `maxSubscribeMetaSize = 1024`, and 2 methods `GetMaxAddrIdentifierSize(height)` and `GetMaxSubscribeMetaSize(height)`, and return corresponding constant depending on height. This way we are separating verification logic (chain/txValidator.go) and constant definition (config.go).

We have to consider that `ChangePoint`(likes `CheckPoint`) not only once in future, I think
`ChangePoint` should be a history Data, not a process routine. For readability and clear eachChange in history.

**Comment by yilunzhang** [comment]
All different GetXXXX() should not be put into the same function, otherwise we will end up with tons of if...else... after a few updates, and it's very hard to see the history of each parameter.
To make the history data explicit and understandable, I think it's better to have something like:
```
GetA(height) {
if (height < xxx) {return xxx}
if (height < xxx) {return xxx}
return a
}

GetB(height) {
if (height < xxx) {return xxx}
if (height < xxx) {return xxx}
return b
}
```
Then we get a very clear and human understandable parameter history.

Conceptually these functions are just extending constants in config.go from static to dynamic, so it's better to put it in config also. Of course we can put it in another file in util/config but still in the same go pkg.

**Comment by gdmmx** [comment]
Now we have two implements:

**Impl_1**
```
GetA(height) {
if (height < xxx) {return xxx}
if (height < xxx) {return xxx}
return a
}

GetB(height) {
if (height < xxx) {return xxx}
if (height < xxx) {return xxx}
return b
}
```
And 
**Impl_2**
```
A.Get(height) {
if (height < config.ChangePoint.A.xxx) {return xxx}
if (height < config.ChangePoint.A.xxx) {return xxx}
return a
}
B.Get(height) {
if (height < config.ChangePoint.B.xxx) {return xxx}
if (height < config.ChangePoint.B.xxx) {return xxx}
return a
}
```

Which is your preference?

**Comment by yilunzhang** [comment]
I think this is probably a more generic way:
```
type HeightDependentInt struct {
  heights []uint32
  values []int
}

func (hdt *HeightDependentInt) ValueAtHeight(height uint32) int {
  return hdt.values[i] if height > hdt.heights[i]
}

A = HeightDependentInt{
  heights: []uint32{100000, 0},
  values: []int{100, 200},
}

A.ValueAtHeight(12345)
```

File: pb/transaction.go (hunk 1 above)
Comment (gdmmx) [reply pb/transaction.go]: Done.

**Comment by NoelBright** [comment]
Looks good to me.

**Comment by gdmmx** [comment]
Done as review suggestion.

**Pull request merged**